ui/src/app/components/runtime-cost-estimator.tsx

```diff
@@ -16,50 +15,41 @@ import { RuntimeConfig } from 'app/utils/runtime-utils';
 import { CSSProperties } from 'react';
 
 interface Props {
-  runtimeParameters: RuntimeConfig;
-  // TODO(RW-7582): remove this prop, this information should be self-contained
-  // by the RuntimeConfig instead.
-  usePersistentDisk: boolean;
+  runtimeConfig: RuntimeConfig;
   costTextColor?: string;
   style?: CSSProperties;
 }
 
+const styles = reactStyles({
+  costSection: {
+    marginRight: '1rem',
+    overflow: 'hidden',
+    whiteSpace: 'nowrap',
+  },
+  cost: {
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+});
+
 export const RuntimeCostEstimator = ({
-  runtimeParameters,
-  usePersistentDisk,
+  runtimeConfig,
   costTextColor = colors.accent,
   style = {},
 }: Props) => {
-  const { computeType, diskSize, pdSize, machine, gpuConfig, dataprocConfig } =
-    runtimeParameters;
-  const {
-    numberOfWorkers = 0,
-    workerMachineType = null,
-    workerDiskSize = null,
-    numberOfPreemptibleWorkers = 0,
-  } = dataprocConfig || {};
-  const workerMachine = findMachineByName(workerMachineType);
-  const gpu = gpuConfig
-    ? findGpu(gpuConfig.gpuType, gpuConfig.numOfGpus)
-    : null;
-  const costConfig = {
-    computeType,
-    masterMachine: machine,
-    gpu,
-    masterDiskSize: usePersistentDisk ? pdSize : diskSize,
-    numberOfWorkers,
-    numberOfPreemptibleWorkers,
-    workerDiskSize,
-    workerMachine,
+  const { computeType, diskSize, detachableDisk } = runtimeConfig;
+  const runningCost = machineRunningCost(runtimeConfig);
+  const runningCostBreakdown = machineRunningCostBreakdown(runtimeConfig);
+  const storageCost = machineStorageCost(runtimeConfig);
+  const storageCostBreakdown = machineStorageCostBreakdown(runtimeConfig);
+  const costStyle = {
+    ...styles.cost,
+    fontSize: detachableDisk ? '12px' : '15px',
```

Comment:
would the font size different just be a time difference issue. Lou changed his font size for that for newer mocks? 
We might be able to use the same style for those cost containers.

Reply:
The cost text overruns its container, causing awkward styling issues if you go beyond a certain value. The mocks are not exhaustive in showing how to deal with large cost values, so the original font was not correct.

Not sure what you mean by time difference.

The font size choice here varies according to detachable disk because including a detachable disk adds a third column to the cost estimator:

![image](https://user-images.githubusercontent.com/822298/151081345-da4342b6-36cd-4741-be76-26ea3bd98f98.png)